### ISSUE TITLE
[Linux] mobi_unpack.py is successful, but XRayBuilder shows error

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Ephemerality/xray-builder#2]
```
I'm trying to get XRayBuilder.exe to run under Linux with Mono. Basically, it 
works, but it doesn't recognise the successful unpacking via mobi_unpack.py or 
even the latest version KindleUnpack.py.

My commandline:
mkdir out
mono XRayBuilder.exe -u ../Mobi_Unpack_v047/lib/mobi_unpack.py -o ./out 
../book.mobi

Output:
X-Ray Builder Version 1.40

Using ./out as an output directory.
Using default Python command. Ensure Python's directory is included in your 
PATH environment variable.
Processing blackout.mobi...
Enter Shelfari URL for blackout (Enter to skip): 
http://www.shelfari.com/books/12345678/book
Running mobi_unpack to get book data...
Error unpacking mobi file: MobiUnpack 0.47
  Copyright (c) 2009 Charles M. Hannum <root@ihack.net>
  With Additions by P. Durrant, K. Hendricks, S. Siebert, fandrieu, DiapDealer, nickredding.
Unpacking Book...
Palm DB type:  BOOKMOBI
Mobi Version:  6
Codec:  utf-8
Title:  MyBook
Palmdoc compression
firstaddl 1c3
ncxidx 1c0
exth flags 50

Processing Mobi format Ebook ...
MetaData from EXTH
{'Publisher': ['My Pub'], 'Language': ['en'], 'Updated Title': ['MyBook'], 
'ASIN': .....}
Unpack raw markup language
Unpacking images, resources, fonts, etc
firstaddl is  451
num_sections is  475
K8Boundary is  -1
Section is  451
Warning: Section 451 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  452
Warning: Section 452 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  453
    extracting image:  image00003.jpeg
Section is  454
Warning: Section 454 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  455
Warning: Section 455 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  456
Warning: Section 456 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  457
Warning: Section 457 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  458
Warning: Section 458 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  459
    extracting image:  image00009.jpeg
Section is  460
    extracting image:  image00010.jpeg
Section is  461
Warning: Section 461 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  462
Warning: Section 462 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  463
Warning: Section 463 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  464
Warning: Section 464 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  465
Warning: Section 465 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  466
Warning: Section 466 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  467
Warning: Section 467 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  468
Warning: Section 468 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  469
Warning: Section 469 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  470
Warning: Section 470 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  471
Warning: Section 471 contains no image or an unknown image format
First 4 bytes: ff d8 ff ed
Section is  472
First 4 bytes: 46 4c 49 53
Skipping  FLIS  section
Section is  473
First 4 bytes: 46 43 49 53
Skipping  FCIS  section
Section is  474
Skip section 474 as it contains the EOF record.
Write ncx
Find link anchors
Insert data into html
Insert hrefs into html
Remove empty anchors from html
Insert image references into html
Error: Referenced image 17 was not recognized as a valid image
Error: Referenced image 18 was not recognized as a valid image
Error: Referenced image 19 was not recognized as a valid image
Error: Referenced image 20 was not recognized as a valid image
Error: Referenced image 1 was not recognized as a valid image
Error: Referenced image 11 was not recognized as a valid image
Error: Referenced image 21 was not recognized as a valid image
Error: Referenced image 5 was not recognized as a valid image
Error: Referenced image 13 was not recognized as a valid image
Error: Referenced image 4 was not recognized as a valid image
Error: Referenced image 7 was not recognized as a valid image
Error: Referenced image 16 was not recognized as a valid image
Error: Referenced image 15 was not recognized as a valid image
Error: Referenced image 12 was not recognized as a valid image
Error: Referenced image 8 was not recognized as a valid image
Error: Referenced image 6 was not recognized as a valid image
Error: Referenced image 14 was not recognized as a valid image
Error: Referenced image 2 was not recognized as a valid image
Write opf
Completed

Press Enter to exit.



So basically, it tells me that there was an error unpacking the book file and 
the error log shows that everything went successfully.
```

Original issue reported on code.google.com by `mbi...@gmail.com` on 26 Nov 2014 at 8:42
